src/loki/worker.cc

```diff
@@ -261,10 +262,11 @@ loki_worker_t::work(const std::list<zmq::message_t>& job,
                     void* request_info,
                     const std::function<void()>& interrupt_function) {
 
-  // grab the request info
+  // grab the request info and make sure to record any metrics before we are done
   auto& info = *static_cast<prime_server::http_request_info_t*>(request_info);
   LOG_INFO("Got Loki Request " + std::to_string(info.id));
   Api request;
+  prime_server::worker_t::result_t result{true, {}, ""};
```

Comment:
i refactored every worker method to look like this. basically i make the result upfront and then overwrite below. this was to allow the same object to carry the final state of the request so that at the bottom of the function i could..